File: 20210829_hash/hash.rb

def dump (a)
  a.each do |key, val|
    p "#{key} => #{val}"
  end
  p "-----"
end

hash = {
  foo: 'bar',
  hoge: 'fuga'
}

p hash
#=> keys are symbol
#
hash[:piyo] = 'piyopiyo'
p hash

hash["piyo"] = 'PIYOPIYO'
dump hash 
#=> "piyo" != :piyo

hash.delete 'piyo'
dump hash

hash.delete :foo
dump hash

hash.each do |a|
  p a
end

File: 20210829_hash/hash.go
package main

// https://ashitani.jp/golangtips/tips_map.html

import "fmt"

func main() {

	hash := map[string]string{"foo": "bar", "hoge": "fuga"}
	fmt.Println(hash)

	hash["piyo"] = "piyopiyo"
	fmt.Println(hash)

	_, ok := hash["foo"]
	fmt.Println(ok)

	_, ok = hash["unknown"]
	fmt.Println(ok)

	delete(hash, "piyo")
	fmt.Println(hash)

	for k, v := range hash {
		fmt.Print(k)
		fmt.Print(" => ")
		fmt.Println(v)
	}

}
